app/app.gradle.kts

```diff
@@ -65,6 +65,70 @@ tasks.withType<Test> {
     dependsOn(project("test_plugins").tasks["jar"])
 }
 
+val nativeShadowTasks = NativePlatforms.values().map { platform ->
+    tasks.create<ShadowJar>("shadowJar-${platform.platformName}") {
+        classifier = platform.platformName
+        configurations = listOf(
+                project.configurations.compile,
+                project.configurations.getByName(platform.platformName)
+        )
+        from(
+                java.sourceSets["main"].output,
+                project(":api").java.sourceSets["main"].output,
+                project(":plugins:base").java.sourceSets["main"].output,
+                project(":plugins:cameraserver").java.sourceSets["main"].output,
+                project(":plugins:networktables").java.sourceSets["main"].output,
+                project(":plugins:powerup").java.sourceSets["main"].output
+        )
+    }
+}
+
+tasks.create("shadowJarAllPlatforms") {
+    nativeShadowTasks.forEach {
+        this.dependsOn(it)
+    }
+}
+
+tasks.withType<ShadowJar> {
```

Comment:
You really want to start using the deferred configuration API.
```kotlin
tasks.withType<ShadowJar>().configureEach {
    ...
```

Reply:
What's the advantage?

Reply:
configureEach uses the lazy task configuration API, which enables quicker builds, and at some point in the future will become the only way to configure tasks.

Reply:
Oh nice :+1: